### DISCUSSION TITLE
SEO behaviour

### DISCUSSION THREAD

**Discussion by http-alba** [globaleaks/globaleaks-whistleblowing-software#3633] [Q&A]
Hi,
We've detected than even though we are enabling the option for search engine indexation in the admin pannel of every tennat, none of them appears in google. Looking through the documentation I'm not able to find the precise information about the behaviour of this feature.
Is there any resource I'm missing or some extra step we should do? What's the expected behaviour in case there is none?
Thanks a lot for your help!

**Comment by evilaliv3** [comment]
Thank you.

Do you have a link where we can look at?

The implementation is simple and the configuration should enable crawlers to index the homepage of the platform.

If you could share some links we may try to give it a look and understand better. Thank you

**Reply by http-alba** [reply]
Some of the last tennants we've configured enabling the SEO that are not visible in search engines:

https://collnargo.bustiaetica.seu-e.cat/
https://comtec.bustiaetica.seu-e.cat/
https://ajripoll.bustiaetica.seu-e.cat/
https://agaur.bustiaetica.seu-e.cat/

**Comment by gingerlime** [comment]
As far as I can see from the robots.txt e.g. https://collnargo.bustiaetica.seu-e.cat/robots.txt indexing is allowed and there's also a sitemap.xml available. You might need to submit those pages to Google or other search engines so it "picks them up". Here's a [helpful article](https://ahrefs.com/blog/submit-website-to-google/) about submitting your website to Google.

**Comment by evilaliv3** [comment]
I confirm what stated by @gingerlime, @http-alba 

The platforms implement all the measures that enables indexing.

The problem is that probably google has not found any site by your clients that is pointing to the platform or if they are listing the platform on some sites they have problem getting indexed.
Can you show us the website of these organization where the platform link is published?

**Comment by gingerlime** [comment]
Good point about creating an external link from a website that's already indexed by Google. This helps Google "discover" other pages and index them. 

Just another small note to add: Google might not see a huge "value" in indexing pages that have little content, and that are nearly-identical to many other pages. In this case, all GlobaLeaks reporting pages are quite similar for Google. So even if it does index these pages, they are unlikely to rank very high on the search result. The more unique "content" there is on the page, the better the chances of Google indexing and ranking it. So if it's important to you for your projects, it might be useful to add some text to explain about the page, and provide further info that Google can also provide to users who search for it. You can add text to the title or Presentation fields for example. 

When it comes to SEO there are very few definitive answers :) Hope this helps.

**Comment by http-alba** [comment]
Thank you all for your previous contributions in the thread. I've been running some URLs through google page speed as it does a SEO audit as well and the engine finds a couple things that could be affecting negatively the SEO:

- _The document does not have a meta description. The description text is empty_: does it have to do with the "description" field that we commonly find in the CMS? Or is it not related?
- _Links are not trackable_: I'm providing an example
`<a href="" data-ng-click="Utils.scrollTo('#Content')" tabindex="1" data-translate="">`

Is there any way we could improve this? I'll be waiting for your advise!

**Comment by evilaliv3** [comment]
Thank you for your question @http-alba 

Actually Google Lighthouse give a SEO score to GlobaLeaks of 100%:
![Screenshot from 2023-09-22 15-44-33](https://github.com/globaleaks/GlobaLeaks/assets/217034/937b73bb-e52f-4261-a634-b4fc7bcde71d)

It is true that it suggest some possible improvements but the score of 100% testimony that are minor improvements and the behaviour of the site is near to perfection when it comes to SEO.

Regarding the specific questions you made:
- problem on the meta description: yes it is related to the description that you configure in the system and i notice that you have set this empty. this surely have bad inpact on seo
- link not trackable. The mentioned link is scrollTo link that can only be implemented in this way so its a false positive that has not impact for seo.

I really think that your clients did not implement correctly their own site and google is struggling to find links to the platform.

If you could provide me the pages where you clients have published the link to the platform we could try to identify what they did wrong.

**Comment by evilaliv3** [comment]
@http-alba : i let you notice that actually clients like callnargo are not publishing any link to the bustia: https://collnargo.ddl.net/ajuntament/canal-de-denuncies

If google do not find the link on their site, there is actually no way google can find the bustia. At google they design web crawlers not magicians! :)

**Comment by gingerlime** [comment]
I completely agree with @evilaliv3 that these scores shouldn't affect Google's ability / willingness to index those sites.

However, to be fair, I do see the SEO issue reported by @http-alba if I use a trial site for example (I observed it also on my own instance)

https://pagespeed.web.dev/analysis/https-seotest-try-globaleaks-org/a44g6m6pdf?form_factor=mobile

it seems to originate from this link (which I don't see on the demo website): `<div id="Skip"><a href="" data-ng-click="Utils.scrollTo('#Content')" tabindex="1" data-translate>Skip to content</a></div>`

<img width="989" alt="CleanShot 2023-09-22 at 16 18 53" src="https://github.com/globaleaks/GlobaLeaks/assets/524274/4a2ba40e-ba7f-4e88-965b-dc6183f626bb">


I don't think it's a major issue, and again, shouldn't prevent crawlers from indexing the site, but it might be worth looking into?

Regarding the `meta` issue, my page has a `meta` tag for `description`: `<meta name="description" content="Secure whistleblowing platform based on GlobaLeaks free and open-source software.">`. Perhaps you deleted the content from the description part of the admin of the site @http-alba ?

**Comment by evilaliv3** [comment]
@gingerlime:

regarding the description i confirm that it has been removed; by restoring it will definitely improve things. @http-alba : You should use the description configuration to describe the site.

regardiong the link: do you have any recommendation for the fix @gingerlime ?

**Comment by gingerlime** [comment]
@evilaliv3 about the description, I thought it might be the case, but if I look at https://pagespeed.web.dev/analysis/https-collnargo-bustiaetica-seu-e-cat/1aza4hjbl1?form_factor=desktop it reports it missing, but when I look at the page source, I see the meta description. This is weird. Looks like a bug in Lighthouse to me. I even found a similar [bug report](https://github.com/GoogleChrome/lighthouse/issues/11659). So let's assume it's a lighthouse bug.

Regarding the link. I'm not too familiar with the codebase yet, but can you help me understand why this `Skip` link appears in some sites, but not on the demo site for example? I'll try to investigate what's the best practice for JS-based links however. I'm pretty sure there's a solution. Maybe just use `href="#"` instead of `href=""` but I'm not 100% sure yet.

**Comment by gingerlime** [comment]
I did a quick test with a static HTML page:

```html
<!doctype html>
<html data-ng-app="GL" data-ng-csp data-ng-strict-di data-ng-keydown="keypress($event)" lang="en">
  <head>
    <meta charset="utf-8" />
    <title>GlobaLeaks</title>
    <meta name="description" content="Secure whistleblowing platform based on GlobaLeaks free and open-source software.">
    <meta name="viewport" content="width=device-width, user-scalable=yes" />

  </head>

  <body class="container-fluid d-flex flex-column" data-ng-class="{{Utils.getCSSFlags()}}" data-flow-prevent-drop>
    <div id="Skip"><a href="" data-ng-click="Utils.scrollTo('#Content')" tabindex="1" data-translate>Skip to content</a></div>
  </body>
</html>
```

Gets the SEO warning from Lighthouse, but if I change only `href="#"` then it passes. I guess it should be an easy fix to make sure all angular links use `href="#"` instead of `href=""` @evilaliv3 ? https://github.com/search?q=repo%3Aglobaleaks%2FGlobaLeaks%20href%3D%22%22&type=code

**Comment by evilaliv3** [comment]
Let's give it a try @gingerlime 

If you would like please feel free to author a patch against the #devel branch!

**Comment by gingerlime** [comment]
@evilaliv3 already working on it :)

**Comment by gingerlime** [comment]
see https://github.com/globaleaks/GlobaLeaks/pull/3651 (not sure if there are tests that might fail, but it should be a minor and valid change)

**Comment by evilaliv3** [comment]
@http-alba: Thanks to @gingerlime we have already improved the SEO score correcting links in relation to Google.

The fix is included in globaleaks 4.13.12

**Comment by http-alba** [comment]
Thank you all for the advise! We usually delete the description by defect so when we deliver the sites to the tennants, they can add their own. I was suspecting this was algo affeting the SEO because most of the times they forget to fill that input or don't see the value. Thank you also for taking in consideration the score correcting for the links. We'll try to work with the tennants from our side and see if there's improvement in the SEO.

**Comment by NickyJr13** [comment]
Sometimes, enabling search engine indexation might not yield immediate results in Google.
Here are a few things to consider:
Google's indexing can take some time, so don't be discouraged if you don't see immediate results.
Ensure that each tenant has high-quality, unique content that provides value to users. Google prioritizes well-structured and informative content.
Make sure you've submitted sitemaps for each tenant to Google Search Console. This helps Google crawl and index your pages efficiently.
Ensure that your tenants' websites are mobile-friendly, as Google considers mobile-friendliness in its rankings.
If you'd like more detailed insights into SEO behavior, you can find more info about professionals in this field from this [source](https://crowdo.net/).
